owncloudApp/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.kt

```diff
@@ -103,52 +92,42 @@ abstract class DrawerActivity : ToolbarActivity() {
+
         if (nav_view != null) {
-            accountChooserToggle = findNavigationViewChildById(R.id.drawer_account_chooser_toogle) as ImageView?
-            accountMiddleAccountAvatar = findNavigationViewChildById(R.id.drawer_account_middle) as ImageView?
-            accountEndAccountAvatar = findNavigationViewChildById(R.id.drawer_account_end) as ImageView?
             // Set background header image and logo, if any
             if (resources.getBoolean(R.bool.use_drawer_background_header)) {
-                (findNavigationViewChildById(R.id.drawer_header_background) as ImageView?)?.setImageResource(R.drawable.drawer_header_background)
+                nav_view.getHeaderView(0).drawer_header_background?.setImageResource(R.drawable.drawer_header_background)
             }
             if (resources.getBoolean(R.bool.use_drawer_logo)) {
                 drawer_logo?.setImageResource(R.drawable.drawer_logo)
             }
 
-            accountChooserToggle?.setImageResource(R.drawable.ic_down)
+            nav_view.getHeaderView(0).drawer_account_chooser_toogle?.setImageResource(R.drawable.ic_down)
             isAccountChooserActive = false
 
             //Notch support
             nav_view.addOnAttachStateChangeListener(object : View.OnAttachStateChangeListener {
                 override fun onViewAttachedToWindow(v: View) {
                     if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
-                        val displayCutout = v.rootWindowInsets.displayCutout
-                        if (displayCutout != null) {
-                            val rlDrawerActiveUser =
-                                findNavigationViewChildById(R.id.drawer_active_user) as ConstraintLayout?
-                            val orientation = resources.configuration.orientation
-                            if (orientation == Configuration.ORIENTATION_PORTRAIT) {
-                                val displayCutoutDP = displayCutout.safeInsetTop /
-                                        (resources.displayMetrics.densityDpi /
-                                                DisplayMetrics.DENSITY_DEFAULT)
-                                rlDrawerActiveUser?.layoutParams?.height =
-                                    resources.getDimension(R.dimen.nav_drawer_header_height).toInt() +
-                                            displayCutoutDP
-                            } else {
-                                rlDrawerActiveUser?.layoutParams?.height =
+                        v.rootWindowInsets.displayCutout?.let {
+                            nav_view.getHeaderView(0).drawer_active_user?.layoutParams?.height =
+                                if (resources.configuration.orientation == Configuration.ORIENTATION_PORTRAIT) {
+                                    val displayCutoutDP =
+                                        it.safeInsetTop / (resources.displayMetrics.densityDpi / DisplayMetrics.DENSITY_DEFAULT)
```

Comment:
This logic (calculate density pixels and similar stuff) fits better in a viewmodel or utils file, activities/fragments should be as dummy as possible, keep it as it is for now and you can move it to the future viewmodel